### PR TITLE
Brad- capture data and mock publishing

### PR DESCRIPTION
**Pull request by hashbrown**
Initial project refactor. Stream parsing is moved into a separate class that outputs each full line, with complete/error/cancel lifecycle hooks. This will encapsulate the protocol as its developed.

Main activity is responsible for handling lines output from the stream, and saving to a file when complete, then scheduling a job request that eventually publishes the file data to the web api.
A JobPublisher is used to support sending data to the web api, it only runs when a network connection is available and attempts to schedule the most efficiently for the device.
Max wait time for a job to run is 1 hour (configurable) provided a network connection exists.  The job will retry until successful and then delete the local file.

### CODE REVIEW

File: Sensei/Sensei.ino [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (justingrammens) [inline Sensei/Sensei.ino]: Good idea to commit the sample .ino file.

**Comment by hashbrown** [comment]
Is it ok to merge this initial work?

**Comment by ps2** [comment]
Yes, looks good to me. Thanks!

**Comment by justingrammens** [comment]
Yep. Sorry for missing this PR before.

---
Founder & Partner
Recursive Awesome | Lab651, LLP
http://recursiveawesome.com | http://lab651.com

**Pull request merged**